### DISCUSSION TITLE
PSRAM initialization on RP2350

### DISCUSSION THREAD

**Discussion by DatanoiseTV** [earlephilhower/arduino-pico#2540] [General]
This is not related to arduino-pico, but I have been working on a Pico-SDK based Audio Development SDK and been trying to port arduino-pico PSRAM to it. I have adopted things from the SDK, added a linker map, but even when I use the linker map with the right parameters, the RP235x will not boot and go into UF2 bootloader mode.

I am using the same linker map:
https://raw.githubusercontent.com/DatanoiseTV/PicoADK-Firmware-Template/refs/heads/RP235x-PD-PSRAM/default.ld.in
https://github.com/DatanoiseTV/PicoADK-Firmware-Template/blob/RP235x-PD-PSRAM/CMakeLists.txt#L150

Any ideas what could be the issue? I am pretty lost there.

**Comment by earlephilhower** [comment]
You're not using the Arduino core, but the linker file you liked has sections for OTA/etc. which push the main app out of its normal spot and you'll not have the ARM-or-RISCV magic number in the right spot, so the ROM will not even try and run things.

Just start with the SDK linker file and add in the PSRAM bits.  It's not much at all...

**Reply by DatanoiseTV** [reply]
Ok, I followed your instructions and wrote some (dirty) integration for it, but PSRAM reports size 0, still.

https://github.com/DatanoiseTV/PicoADK-Firmware-Template/tree/PicoADK-RP235x-Bare-PSRAM

Is the PSRAM initialization code bit called automatically by the Pico-SDK or do I need to call some init function manually?
I see the macro PICO_RUNTIME_INIT_FUNC_RUNTIME and as far as I understand it should be called?
